### PR TITLE
Wikiconv v2 -- Reconstruction Pipeline Fixes

### PR DESCRIPTION
**Pull request by vegetable68**
This PR:

- Added testing functionalities to the conversation reconstructor.
- Tested on two previously failed revisions.
    - Fixed memory issues in previous pipeline (package mwparserfrom hell is unreliable).
    - Changed double encoded fields.
- Changed action names according to the paper.

### CODE REVIEW

File: conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py [hunk 1]
```diff
@@ -39,30 +36,28 @@ def isheading(line):
     for x in line:
         if x == '=':
            front_cnt += 1
-    for x in line[::-1]: 
+    for x in line[::-1]:
```
Comment (sorensenjs) [inline conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py]: are you counting the number of '=' characters? len(re.findall('=', line))

File: conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py [hunk 2]
```diff
@@ -157,8 +156,8 @@ def locate_new_token_pos(old_pos, ops, errorchoice='raise_error'):
                         else:
                             new_pos = op['b1']
             if old_pos == op['a2']:
-                if errorchoice == 'left_bound':# and not(op['tokens'][-1].type == 'break' and op['name'] == 'insert'):
+              if errorchoice == 'left_bound' and op['name'] == 'insert':
                    new_pos = op['b1']
-                else:
-                   new_pos = max(new_pos, op['b2'])
+              if not(errorchoice == 'left_bound'):
```
Comment (sorensenjs) [inline conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py]: !=

File: conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/third_party/rev_clean.py [hunk 3]
```diff
@@ -68,29 +65,27 @@ def remove_date(comment):
...
-    ret = strip_html(ret)
-    ret = substitute_patterns(ret, post_sub_patterns)
+    # Change format for better diff
     ret = re.sub('[\n]+', '\n', str(ret))
     ret = '\n'.join([x.strip() for x in ret.splitlines() if not(x.strip() == "")]) + '\n'
     if ret == '\n': return ""
     return ret
+
+def clean(rev):
+    ret = rev
```
Comment (sorensenjs) [inline conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/third_party/rev_clean.py]: do ret = str(rev)
if you need it

File: conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py (hunk 1 above)
Comment (vegetable68) [reply conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py]: No only the consecutive '='s at the beginnging and end of the line (as a wikipedia format).

Comment (sorensenjs) [reply conversation_reconstruction/dataflow/conversation_reconstruction/construct_utils/utils/insert_utils.py]: but there's no break statement, you could do len(re.find('^=+').group(0))
or something like that.

**Pull request merged**